internal/pkg/iostreams/io.go

```diff
@@ -156,7 +161,46 @@ func (s *system) ReadSecret() ([]byte, error) {
 		return nil, fmt.Errorf("prompting is disabled")
 	}
 
-	return term.ReadPassword(int(s.in.Fd()))
+	fd := int(s.in.Fd())
+
+	// Store and restore the terminal status on interruptions to
+	// avoid that the terminal remains in the password state
+	// This is necessary as for https://github.com/golang/go/issues/31180
+	oldState, err := term.GetState(fd)
+	if err != nil {
+		return make([]byte, 0), err
```

Comment:
```suggestion
		return nil, err
```

Nil should be fine on error